src/main/java/org/jenkinsci/plugins/parallel_test_executor/ParallelTestExecutor.java

```diff
@@ -194,10 +211,14 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
                 boolean shouldIncludeElements = generateInclusions && i != 0;
                 List<String> elements = new ArrayList<String>();
                 r.add(new InclusionExclusionPattern(elements, shouldIncludeElements));
-                for (TestClass d : sorted) {
+                for (TestEntity d : sorted) {
                     if (shouldIncludeElements == (d.knapsack == k)) {
-                        elements.add(d.getSourceFileName(".java"));
-                        elements.add(d.getSourceFileName(".class"));
+                        if (!isCaseCentric) {
+                            elements.add(d.getOutputString(".java"));
+                            elements.add(d.getOutputString(".class"));
+                        } else {
+                            elements.add(d.getOutputString(""));
```

Comment:
Seems weird. (This argument is ignored.) Would make more sense to me for the `TestEntity` to return a `List<String>` of outputs. `TestCase` would return `singleton(output)` whereas `TestClass` would return a pair of names.